### PR TITLE
[AIR][Docs] Fix torch_image_example

### PR DESCRIPTION
**Pull request by jiaodong**
## Why are these changes needed?

This example failed for a while due to changes made while the notebook is not covered by CI.

## Notable changes:

In order for training prediction to be consistent, we need to make images looks like `TensorArray([])` rather than `[TensorArray]`.
```python
def convert_batch_to_pandas(batch: Tuple[torch.Tensor, int]) -> pd.DataFrame:
    images = TensorArray([image.numpy() for image, _ in batch])
    labels = [label for _, label in batch]

    df = pd.DataFrame({"image": images, "label": labels})

    return df

train_dataset = train_dataset.map_batches(convert_batch_to_pandas)
test_dataset = test_dataset.map_batches(convert_batch_to_pandas)
```

Make changes to `batch_predictor.predict` since fields like `feature_column` is deprecated. Also drop the "label" column in dataset we used for prediction

```python
from ray.train.torch import TorchPredictor
from ray.train.batch_predictor import BatchPredictor

predict_dataset = test_dataset.drop_columns(cols=["label"])
batch_predictor = BatchPredictor.from_checkpoint(
    checkpoint=latest_checkpoint,
    predictor_cls=TorchPredictor,
    model=Net(),
)

outputs: ray.data.Dataset = batch_predictor.predict(data=predict_dataset, dtype=torch.float)
```

Add a custom http_adapter to map our input Ndarray to correct float32 type otherwise we will read double by default.

```python
from ray import serve
from ray.serve import PredictorDeployment
from ray.serve.http_adapters import NdArray

def json_to_numpy(payload: NdArray) -> pd.DataFrame:
      """Accepts an NdArray JSON from an HTTP body and converts it to a Numpy Array."""
      # Have to explicitly convert to float since np.array reads as a double.
      arr = np.array(payload.array, dtype=np.float32)
      return arr

serve.start(detached=True)
deployment = PredictorDeployment.options(name="my-deployment")
deployment.deploy(TorchPredictor, latest_checkpoint, batching_params=False, model=Net(), http_adapter=json_to_numpy)
```

## Test plan

`bazel test //doc/source/ray-air/examples:torch_image_example` passed on go/repro-ci on https://buildkite.com/ray-project/ray-builders-pr/builds/37656#0181df54-cf8e-456a-bd88-1026c2558400 that previously failed.

## Related issue number

#26410

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by xwjiang2010** [comment]
Thanks @jiaodong,
do you want to uncomment it here? 
`doc/source/ray-air/examples/BUILD`

**Review by xwjiang2010 — APPROVED** [review]

**Comment by amogkam** [comment]
Instead of implementing the custom adapter, I think you can specify a dtype as a predict_kwarg instead.

File: doc/source/ray-air/examples/torch_image_example.ipynb
Comment (richardliaw) [inline doc/source/ray-air/examples/torch_image_example.ipynb:24]: nit - clear output?

**Comment by jiaodong** [comment]
> ```python
> PredictorDeployment
> ```

hmm you mean something like this ?

```
deployment.deploy(TorchPredictor, latest_checkpoint, batching_params=False, model=Net(), predict_kwargs={"dtype": np.float32})
```

I got some JSONDecodeError :/

**Review by amogkam — APPROVED** [review]
Ok approving for now to fix the regression, but we should fix the adapter issue. 

The default adapters should work, and users should not need to write their own.

**Comment by jiaodong** [comment]
@amogkam I think we can give that a try but I made a few attempts and haven't found a default http adapter that works out of the box e2e yet :/

**Comment by amogkam** [comment]
@jiaodong do you think you can reproduce and file an issue? If the default workflow is broken then I think we should fix before beta 😅

**Comment by amogkam** [comment]
(but for this particular PR, let's merge it asap. Just need codeowner approval(

**Comment by jiaodong** [comment]
@amogkam https://github.com/ray-project/ray/issues/26539 filed

**Review by richardliaw — APPROVED** [review]

**Pull request merged**
